test/suites/standard/log-scan.robot

```diff
@@ -0,0 +1,49 @@
+*** Settings ***
+Documentation       Tests for various log messages we do or do not want.
+
+Resource            ../../resources/common.resource
+Resource            ../../resources/microshift-process.resource
+Library             ../../resources/journalctl.py
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           restart    slow
+
+
+*** Variables ***
+${CURSOR}       ${EMPTY}    # The journal cursor before restarting MicroShift
+
+
+*** Test Cases ***
+Should Not Find Forbidden
+    [Documentation]    Verify that the logs for the current run of MicroShift do not include the word "forbidden"
+    Pattern Should Not Appear In Log Output    ${CURSOR}    forbidden
+
+Should Not Find Cannot Patch Resource
+    [Documentation]    Verify that the logs for the current run of MicroShift do not include the words "cannot patch resource"
+    Pattern Should Not Appear In Log Output    ${CURSOR}    cannot patch resource
+
+Should Find Etcd Is Ready
+    [Documentation]    Verify that the logs for the current run of MicroShift include the words "etcd is ready"
+    Pattern Should Appear In Log Output    ${CURSOR}    etcd is ready
+
+Should Find MicroShift Is Ready
+    [Documentation]    Verify that the logs for the current run of MicroShift include the words "microshift is ready"
+    Pattern Should Appear In Log Output    ${CURSOR}    microshift is ready
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig    # for readiness checks
+    ${cursor}=    Get Journal Cursor
+    Set Suite Variable    \${CURSOR}    ${cursor}
+    Restart MicroShift
```

Comment:
Is this sequence of actions used to also get shutdown logs?

Reply:
Honestly, I was just trying to avoid writing separate stop and start keywords, but you make a good point. It will have the effect of capturing the shutdown and startup messages so I've added a [retcon](https://en.wikipedia.org/wiki/Retroactive_continuity) comment indicating that it's on purpose.